typing/typedecl.ml

```diff
@@ -1225,9 +1225,18 @@ let check_constraints env sdecl (_, decl) =
    immediate, we should check the manifest is immediate). Also, update the
    resulting jkind to match the manifest. *)
 let narrow_to_manifest_jkind env loc decl =
-  match decl.type_manifest with
-  | None -> decl
-  | Some ty ->
+  match decl.type_manifest, decl.type_kind with
+  | None, _ -> decl
+  | Some _, (Type_record _ | Type_record_unboxed_product _ | Type_variant _ | Type_open)
+    when not (Builtin_attributes.has_or_null_reexport decl.type_attributes)
```

Comment:
This condition should be remarked on in the (helpful!) comment.

Reply:
I'm going to punt on this. How or_null is handle is going to have to change after the rebase, so this code is going to change